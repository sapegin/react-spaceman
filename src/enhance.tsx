import * as React from 'react';
import { Size } from './consts';
import { Panel } from './Panel';

interface IProps {
	below?: Size,
	inset?: Size,
	x?: Size,
	y?: Size,
	between?: Size,
	inline?: boolean,
	squish?: boolean,
	className?: string,
	children?: React.ReactNode,
	[key: string]: any,
}

interface IDefaults {
	below?: Size,
	inset?: Size,
	x?: Size,
	y?: Size,
	between?: Size,
	inline?: boolean,
	squish?: boolean,
	className?: string,
}

export default function(tag: React.ReactType, defaults: IDefaults = {}) {
	return ({ below, inset, x, y, between, squish, className, children, ...props }: IProps) => (
		<Panel
			tag={tag}
			tagProps={props}
			below={below || defaults.below}
			inset={inset || defaults.inset}
			x={x || defaults.x}
			y={y || defaults.y}
			between={between || defaults.between}
			squish={squish || defaults.squish}
			className={className || defaults.className}
		>
			{children}
		</Panel>
	);
}
