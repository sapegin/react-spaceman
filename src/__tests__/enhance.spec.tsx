import { shallow } from 'enzyme';
import * as React from 'react';
import enhance from '../enhance';

describe('enhance', () => {
	it('should return a function', () => {
		const result = enhance('div');
		expect(result).toEqual(expect.any(Function));
	});

	it('returned function should render a div', () => {
		const Div = enhance('div');
		const wrapper = shallow(<Div>hello</Div>);
		expect(wrapper).toMatchSnapshot();
	});

	it('returned function should accept Spaceman props', () => {
		const Div = enhance('div');
		const wrapper = shallow(<Div below="m">hello</Div>);
		expect(wrapper).toMatchSnapshot();
	});

	it('returned function should set default Spaceman props', () => {
		const Div = enhance('div', { below: 'm' });
		const wrapper = shallow(<Div>hello</Div>);
		expect(wrapper).toMatchSnapshot();
	});

	it('returned function should pass extra props to a div', () => {
		const Div = enhance('div');
		const wrapper = shallow(<Div id="pizza">hello</Div>);
		expect(wrapper).toMatchSnapshot();
	});
});
