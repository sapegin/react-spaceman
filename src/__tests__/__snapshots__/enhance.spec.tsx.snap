// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enhance returned function should accept Spaceman props 1`] = `
<Panel
  below="m"
  tag="div"
  tagProps={Object {}}
>
  hello
</Panel>
`;

exports[`enhance returned function should pass extra props to a div 1`] = `
<Panel
  tag="div"
  tagProps={
    Object {
      "id": "pizza",
    }
  }
>
  hello
</Panel>
`;

exports[`enhance returned function should render a div 1`] = `
<Panel
  tag="div"
  tagProps={Object {}}
>
  hello
</Panel>
`;

exports[`enhance returned function should set default Spaceman props 1`] = `
<Panel
  below="m"
  tag="div"
  tagProps={Object {}}
>
  hello
</Panel>
`;
